ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -728,6 +753,29 @@ void setChunkException(RouterException exception) {
     }
 
     /**
+     * Slice this chunk's data to only include the bytes within the operation's specified byte range.
+     * @param blobData the {@link BlobData} for this chunk.
+     * @return A {@link ByteBuffer} that only includes bytes within the operation's specified byte range.
+     */
+    protected ByteBuffer filterChunkToRange(BlobData blobData) {
+      ByteBuffer buf = blobData.getStream().getByteBuffer();
+      if (options == null || options.getRange() == null) {
+        return buf;
+      }
+      if (resolvedByteRange.getRangeSize() == 0) {
```

Comment:
Yes, in the case of a simple blob, the chunk would have already been fetched and this method will be called to filter the byte range.  However, in the case of a composite blob, the list of chunks to fetch will be empty, so the operation will be completed immediately.